exchanges/graphcache/src/ast/schemaPredicates.ts

```diff
@@ -111,3 +112,20 @@ function expectAbstractType(
     5
   );
 }
+
+export function expectValidKeyingConfig(
+  schema: GraphQLSchema,
+  keys: KeyingConfig
+): void {
+  const types = Object.keys(schema.getTypeMap());
```

Comment:
Awesome! Didn’t realise it’d be this simple!
All this is missing is a quick `if (process.env.NODE_ENV !== “production”)` check so this function compiles away.
Afaik, empty function bodies (since `warn` does compile away) may not be optimised away as the property reads themselves could have side-effects?

I’m not sure about the above, but we can simply check the production min bundle for this `Object.keys` signature.

Also, I think `warn` instead of `invariant` is perfect! We have quite a lot of warnings in Graphcache for unexpected conditions that _won’t_ stop it from working. A misnamed configuration item would be part of that imho. Especially since the schema can go out of sync with the config/queries in development.

Reply:
The Webpack/Rollup optimisations are very cool! Have added that `if NODE_ENV` and the new code is now removed in `.min.js`:

![image](https://user-images.githubusercontent.com/398210/82513362-b1038e80-9b0a-11ea-912d-44c80d844e5f.png)